apps/form.go

```diff
@@ -15,17 +19,15 @@ package apps
 // autocomplete once the subcommand is selected is designed to mirror the
 // functionality of the Modal. Some gaps and differences still remain.
 //
-// Requests for forms are calls, can use Expand, making it easy to generate
-// forms specific to the user, channel, etc.
-//
-// When a dynamic select field is selected in a Modal, or in Autocomplete, a
-// Lookup call request is made to the Form's Call. The app should respond with
-// "data":[]SelectOption, and "type":"ok".
-//
-// When a select field with "refresh" set changes value, it forces reloading of
-// the form. A call request type form is made to fetch it, with the partial
-// values provided. Expected response is a "type":"form" response.
+// A form can be dynamically fetched if it specifies its Source. Source may
+// include Expand and State, allowing to create custom-fit forms for the
+// context.
 type Form struct {
+	// Source is the call to make when the form's definition is required (i.e.
+	// it has no fields, or needs to be refreshed from the app). A simple call
+	// can be specified as a path (string). It will contain no expand/state.
```

Comment:
"It will contain no expand/state." - we mention above that "Source may include Expand and State" ?